lib/widgets/message_list.dart

```diff
@@ -57,19 +57,35 @@ class _MessageListState extends State<MessageList> {
   @override
   Widget build(BuildContext context) {
     assert(model != null);
-    if (!model!.fetched) return const Center(child: CircularProgressIndicator());
+    if (!model!.fetched)
+      return const Center(child: CircularProgressIndicator());
 
     return DefaultTextStyle(
         // TODO figure out text color -- web is supposedly hsl(0deg 0% 20%),
         //   but seems much darker than that
         style: const TextStyle(color: Color.fromRGBO(0, 0, 0, 1)),
         child: ColoredBox(
             color: Colors.white,
-            child: Padding(
-                padding: const EdgeInsets.symmetric(horizontal: 8),
-                child: ConstrainedBox(
-                    constraints: const BoxConstraints(maxWidth: 760),
-                    child: _buildListView(context)))));
+
+            // Pad the left and right insets, for small devices in landscape.
+            child: SafeArea(
+                // A non-ancestor (the compose box) pads the bottom inset;
+                // prevent extra padding here.
+                bottom: false,
+
+                // A non-ancestor (the app bar) pads the top inset. But no
+                // need to prevent extra padding with `top: false`, because
+                // Scaffold, which knows about the app bar, sets `body`'s
+                // ambient `MediaQueryData.padding` to have `top: 0`:
+                //   https://github.com/flutter/flutter/blob/3.7.0-1.2.pre/packages/flutter/lib/src/material/scaffold.dart#L2778
```

Comment:
Given how convenient it is that we don't need to worry about the app bar and disabling the top inset, I suspect the preferred Flutter style here would be for MessageListPage to do its children the same sort of courtesy that Scaffold is doing. Then we'd get to just say `SafeArea(minimum: …, child: …)` and have that do the right thing, with no worrying about what siblings or neighbors this widget might have.

Especially because in the future we might have message lists without a compose box, just as we do in the RN app.

No need to do that right now, but I'll add a TODO comment with this thought.